lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1255,8 +1256,10 @@ static bool performCompile(CompilerInstance &Instance,
     // Disable delayed parsing of type and function bodies when we've been
     // asked to dump the resulting AST.
     bool CanDelayBodies = Action != FrontendOptions::ActionType::DumpParse;
-    Instance.performParseOnly(/*EvaluateConditionals*/
-                    Action == FrontendOptions::ActionType::EmitImportedModules,
+    bool EvaluateConditionals =
+        Action == FrontendOptions::ActionType::EmitImportedModules
+        || Action == FrontendOptions::ActionType::ScanDependencies;
+    Instance.performParseOnly(EvaluateConditionals,
```

Comment:
The flow in here and through 1271 is confusing. The predicate here is `FrontendOptions::shouldActionOnlyParse(Action)` and I would expect a return right after line 1262.
But instead, it falls through to 1271: `if (Action == FrontendOptions::ActionType::Parse)` and it that is true it returns. Which leaves me pondering: what's the difference between the two predicates? Is one a subset of the other? There must be a clearer way to write this. Maybe also do a return right here?
OK, I see it, now. Will leave the above confused ramblings in FWIW.

The whole business starting at 1255 could use a comment to the effect of do the the right about of parsing, etc. for the mode, or some such. (I would factor it out into a function with that name.)

Then, a blank line after 1258 would make it clear that the comment above it only applies to that one line.
(Not your fault!)

Then, a "why" comment might be nice about why conditionals are only being evaluated in these two cases.

I guess what's going on is that this function could be split in a "parse" phase, through 1272 and an after-parse phase, and that demarcation would clarify what's going on. (Not your fault, but this change just makes it incrementally worse, like an amoeba that's overdue for fission.)